CMakeLists.txt

```diff
@@ -34,152 +44,210 @@ option(UNDEFINED_BEHAVIOR_SANITIZER "Build with UndefinedBehaviorSanitizer." OFF
 option(ALIGNED_MEMORY_MODEL "Aligned memory model ONLY." OFF)
 option(SET_SSL_CALLBACKS "Set SSL thread and lock callbacks." OFF)
 
-set(TYPE_OF_LIB SHARED)
+# TODO: Use target_include_directories to not give headers to WHOLE project
+#   - Print out the dependencies for each target via command line and verify there are no duplicate libraries being linked
+
+
 if (BUILD_STATIC)
   set(TYPE_OF_LIB STATIC)
+else()
+  set(TYPE_OF_LIB SHARED)
 endif()
 
-set(CMAKE_MACOSX_RPATH TRUE)
-
-get_filename_component(ROOT "${CMAKE_CURRENT_SOURCE_DIR}" ABSOLUTE)
-
-add_definitions(-DVERSION_STRING=\"${PROJECT_VERSION}\")
-
 if(CONSTRAINED_DEVICE)
-    add_definitions(-DCONSTRAINED_DEVICE)
+  add_definitions(-DCONSTRAINED_DEVICE)
 endif()
 
+set(CMAKE_MACOSX_RPATH TRUE)
+add_definitions(-DVERSION_STRING=\"${PROJECT_VERSION}\")
+
+# Set where to build this project.
 if(NOT KINESIS_VIDEO_PRODUCER_C_SRC)
-    if(DEFINED ENV{KINESIS_VIDEO_PRODUCER_C_SRC})
-        set(KINESIS_VIDEO_PRODUCER_C_SRC $ENV{KINESIS_VIDEO_PRODUCER_C_SRC})
-    else()
-        set(KINESIS_VIDEO_PRODUCER_C_SRC "${CMAKE_CURRENT_SOURCE_DIR}")
-    endif()
+  if(DEFINED ENV{KINESIS_VIDEO_PRODUCER_C_SRC})
+    set(KINESIS_VIDEO_PRODUCER_C_SRC $ENV{KINESIS_VIDEO_PRODUCER_C_SRC})
+  else()
+    set(KINESIS_VIDEO_PRODUCER_C_SRC "${CMAKE_CURRENT_SOURCE_DIR}")
+  endif()
 endif()
-
 message(STATUS "Kinesis Video Producer path is ${KINESIS_VIDEO_PRODUCER_C_SRC}")
 
+# NOTE: The windows CI build has CMAKE_BUILD_TYPE set to debug, probably due to MSVC environment, TODO: look into this
+# Default to release build if unspecified.
 if(NOT CMAKE_BUILD_TYPE)
-    message(STATUS "Setting CMAKE_BUILD_TYPE to Release by default")
-    set(CMAKE_BUILD_TYPE "Release" CACHE STRING "Choose the type of build." FORCE)
+  message(STATUS "Setting CMAKE_BUILD_TYPE to Release by default")
+  set(CMAKE_BUILD_TYPE "Release" CACHE STRING "Choose the type of build." FORCE)
+else()
+  message(STATUS "CMAKE_BUILD_TYPE is set to: ${CMAKE_BUILD_TYPE}")
 endif()
 
+
+# Fetch and build 3rd-party dependencies if specified.
 if(BUILD_DEPENDENCIES)
+  message(STATUS "Begin building dependencies.")
+
+  # Set directory where to build the dependencies.
   if (NOT OPEN_SRC_INSTALL_PREFIX)
-    set(OPEN_SRC_INSTALL_PREFIX ${CMAKE_CURRENT_SOURCE_DIR}/open-source)
+    set(OPEN_SRC_INSTALL_PREFIX ${CMAKE_SOURCE_DIR}/open-source)
```

Comment:
I think it should be `${CMAKE_CURRENT_SOURCE_DIR}`? Did we test webrtc with this build to confirm it works and does not break existing behavior if `OPEN_SRC_INSTALL_PREFIX` is not set?